vm/interpreter/src/circ_supply.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use actor::*;
+use address::Address;
+use blocks::Tipset;
+use chain::*;
+use cid::Cid;
+use clock::ChainEpoch;
+use fil_types::{FILECOIN_PRECISION, UPGRADE_IGNITION_HEIGHT, UPGRADE_LIFTOFF_HEIGHT};
+use ipld_blockstore::BlockStore;
+use num_bigint::BigInt;
+use state_tree::StateTree;
+use std::collections::HashMap;
+
+#[derive(Default)]
+pub struct GenesisInfo {
+    genesis_msigs: Vec<multisig::State>,
+    // info about the Accounts in the genesis state
+    genesis_actors: Vec<GenesisActor>,
+    genesis_pledge: TokenAmount,
+    genesis_market_funds: TokenAmount,
+}
+
+pub struct GenesisActor {
+    addr: Address,
+    init_bal: TokenAmount,
+}
+
+fn get_actor_state<DB: BlockStore>(
+    state_tree: &StateTree<DB>,
+    addr: &Address,
+) -> Result<ActorState, String> {
+    state_tree
+        .get_actor(&addr)
+        .map_err(|_| "failed to get actor".to_string())?
+        .ok_or_else(|| "Actor address does not exist".to_string())
+}
+
+pub fn get_fil_vested<DB: BlockStore>(
+    pre_ignition: &GenesisInfo,
+    post_ignition: &GenesisInfo,
+    height: ChainEpoch,
+    state_tree: &StateTree<DB>,
+) -> Result<TokenAmount, String> {
+    let mut return_value = TokenAmount::default();
+
+    if height <= UPGRADE_IGNITION_HEIGHT {
+        for actor in &pre_ignition.genesis_msigs {
+            return_value += &actor.initial_balance - actor.amount_locked(height);
+        }
+    } else {
+        for actor in &post_ignition.genesis_msigs {
+            return_value +=
+                &actor.initial_balance - actor.amount_locked(height - actor.start_epoch);
+        }
+    }
+
+    for actor in &pre_ignition.genesis_actors {
+        let state = get_actor_state(state_tree, &actor.addr)?;
+        let diff = &actor.init_bal - state.balance;
+        if diff > TokenAmount::default() {
+            return_value += diff
+        }
+    }
+
+    return_value += &pre_ignition.genesis_pledge + &pre_ignition.genesis_market_funds;
+    Ok(return_value)
+}
+
+pub fn get_fil_mined<DB: BlockStore>(state_tree: &StateTree<DB>) -> Result<TokenAmount, String> {
+    let reward_actor = get_actor_state(state_tree, &*REWARD_ACTOR_ADDR)?;
+    let reward_state: reward::State = state_tree
+        .store()
+        .get(&reward_actor.code)
```

Comment:
you're loading the actor code? There is nothing in store for this

Reply:
My bad that should be reward_actor.state